cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -430,9 +430,14 @@ private DatasetTypeMeta getTypeInfo(NamespaceId namespaceId, String typeName) th
     try {
       return typeService.getType(datasetTypeId);
     } catch (DatasetTypeNotFoundException e) {
-      // Type not found in the instance's namespace. Now try finding it in the system namespace
-      DatasetTypeId systemDatasetTypeId = ConversionHelpers.toDatasetTypeId(NamespaceId.SYSTEM, typeName);
-      return typeService.getType(systemDatasetTypeId);
+      try {
+        // Type not found in the instance's namespace. Now try finding it in the system namespace
+        DatasetTypeId systemDatasetTypeId = ConversionHelpers.toDatasetTypeId(NamespaceId.SYSTEM, typeName);
+        return typeService.getType(systemDatasetTypeId);
+      } catch (DatasetTypeNotFoundException e1) {
```

Comment:
what's the point of wrapping these 2 lines of code w/ try-catch block?
Seems to have the same effect as without wrapping.